src/lib.rs

```diff
@@ -29,15 +30,55 @@ mod dlmalloc;
 #[cfg(all(feature = "global", not(test)))]
 mod global;
 
+/// A platform interface
+pub trait System: Send {
```

Comment:
Oh as a side note this should probably be `unsafe` due to it having various invariants about memory lifetimes to uphold.

Additionally I think that after the trait is `unsafe` the methods themselves probably don't have to be `unsafe`?